### PR TITLE
Fixing integration in latest raml2obj and api-console

### PR DESCRIPTION
**Pull request by antoniogarrote**
Fixing problem with inheritance from non-canonical base type and computing minType of a union inheriting from a non-union.

We were not canonilising the base type when inheriting.
This can lead to situations where the base type is a union (e.g. union in a property in a object type).
The additional clause in the minType function deals with unions inheriting from non-nunions.

Missing test, errors were coming from using versino 2.5 in raml2obj. Test cases can be taken from there.

### CODE REVIEW

**Comment by jstoiko** [comment]
@trevorr since this relates to some of the issues you've been having, would you mind taking a look?

**Comment by jstoiko** [comment]
> Missing test, errors were coming from using versino 2.5 in raml2obj. Test cases can be taken from there.
@antoniogarrote can you provide a link to those test cases?

**Comment by ForsakenHarmony** [comment]
Could you add a test case for the new section?

**Comment by jarrodek** [comment]
Guys, I don't want to rush you but there's A LOT of people are waiting for this to be merged :)

**Comment by jarrodek** [comment]
Can anyone tell me why sometimes type definition is like this:
```
"name": "Entity",
"displayName": "Entity",
"type": {
  "type": "object"
},
"required": true,
"properties": ...
```
And sometimes like this:

```
"name": "Product",
"type": "object",
"properties": ...
```

The `type` property is sometimes an object and sometimes a string.

Also, why sometimes I see this as a type definition:

```
"name": "Head",
 "displayName": "Manager",
"type": "ExampleType.Person",
```
Type should be `object` and not `ExampleType.Person`. It is already expanded..

**Comment by trevorr** [comment]
I'm taking a look, but right away I notice that this would reintroduce #65, where any extra attributes of the supertype would be lost in the new minType case.

File: src/minType.js
Comment (trevorr) [inline src/minType.js]: Should these arguments be reversed? The signature is `minType(sup, sub)`.

**Comment by trevorr** [comment]
I also wonder if minType 7, 8, and this new case ("9") could/should be combined into a unified `supType === 'union' || subType === 'union'` case that generically permutes the union members with recursive calls to minType. In other words, if case 7 were generalized to treat a non-union super- or sub-type as a union of a single member, they could all collapse into one case. That also addresses the fact that case "9" is currently missing restriction and consistency checks (as well as not preserving extra attributes of the super-type).

**Comment by jarrodek** [comment]
Sorry, forget about this `"type": "ExampleType.Person",`. It's the same is it was in old version.
But `type: {type: 'object'}` is just redundant.

**Comment by jarrodek** [comment]
I've made an adjustment in raml2obj to transform this property to `type: object`. So it's not a high priority.

**Comment by jarrodek** [comment]
Hey, what's the ETA for this PR to be merged?

**Comment by jstoiko** [comment]
waiting on @antoniogarrote’s answer re: Trevor’s comment

File: src/minType.js
Comment (antoniogarrote) [reply src/minType.js]: Fixed

**Comment by antoniogarrote** [comment]
Updated the commit

**Comment by ForsakenHarmony** [comment]
Not gonna add a test case?

**Comment by jstoiko** [comment]
Let's bring the discussion to Slack to make sure we're all on the same page before merging. I invited all of you except for @trevorr: would you mind [adding yourself](https://join.slack.com/t/raml-api/shared_invite/enQtMjc2ODQwNjA0Mjc4LWU1NzdmYmFkZTdjZmE0MmY3Y2JmNWU4Yjk3ZWRmMjE4ZjE0NmIxOTc0MDViYTRhYjIyNGY3YjY5MWM3MTVlYmQ)?

File: src/minType.js
Comment (trevorr) [inline src/minType.js]: Style-wise, perhaps this should be a for-of loop, as is done elsewhere in this function?

File: src/minType.js
Comment (trevorr) [inline src/minType.js]: Is it safe to modify the given object? Elsewhere in this function we create a new object.

**Comment by trevorr** [comment]
I've pushed my changes for generalizing union handling in minType. They include the related change in #82, so that PR can be closed if this is merged.

**Comment by trevorr** [comment]
I've also verified that all the raml2obj tests pass with this PR.

**Comment by jstoiko** [comment]
Nice. on my end, I need to make sure all the [ramldt2jsonschema](https://github.com/raml-org/ramldt2jsonschema) tests pass, then I'll merge. Thanks again @trevorr !

**Review by jstoiko — APPROVED** [review]

**Pull request merged**
